libs/orbit-essentials/src/utils/random.rs

```diff
@@ -19,23 +19,26 @@ thread_local! {
     target_vendor = "unknown",
     target_os = "unknown"
 ))]
+#[no_mangle]
 /// A getrandom implementation that works in the IC.
-pub fn custom_getrandom_bytes_impl(dest: &mut [u8]) -> Result<(), getrandom::Error> {
+unsafe extern "Rust" fn __getrandom_v03_custom(
+    dest: *mut u8,
+    len: usize,
+) -> Result<(), getrandom::Error> {
     RNG.with(|rng| {
+        let buf = unsafe {
+            // fill the buffer with zeros
+            core::ptr::write_bytes(dest, 0, len);
```

Comment:
why do we need to zero it if we almost immediately overwrite it?

Reply:
I was following [this](https://docs.rs/getrandom/0.3.1/getrandom/index.html#custom-backend) example. I guess the reason is to have predictable output in case the function called subsequently did not write all bytes.

Reply:
I understand but that wouldn't be pretty problematic behaviour to not return the required random bytes, wouldn't it? 😅

Reply:
It's not about being more random, just that the fill_bytes should completely fill the buffer, otherwise this method's behavior is undefined. 

According to the docs, this behavior should be guaranteed: https://docs.rs/rand/latest/rand/trait.RngCore.html#tymethod.fill_bytes

If it fills all bytes then zeroing the buffer out beforehand is wasted instructions.